### PR TITLE
feat: thousand separator on value input

### PR DESCRIPTION
**Pull request by thr0wn**
- New component <InputNumber /> to enhance the experience of typing numbers
- Use InputNumber on CreateToken, TokenMint, TokenMelt and SendTokens

### CODE REVIEW

File: src/components/InputNumber.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+  onValueChange: PropTypes.func,
+  /**
+   * Same behavior of React input onChange
+   */
+  onChange: PropTypes.func,
+};
+
+InputNumber.defaultProps = {
+  defaultValue: "",
+  precision: 2,
```
Comment (obiyankenobi) [inline src/components/InputNumber.js:149]: `hathorLib.constants.DECIMAL_PLACES`

**Comment by pedroferreira1** [comment]
It would be good to have a GIF with the new input component working

File: src/components/InputNumber.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      const decimalPart = Intl.NumberFormat(locale).format(
+        value.substr(0, value.length - precision)
+      );
+      const fractionalPart = value.substr(value.length - precision);
+      return `${decimalPart}${separator}${fractionalPart}`;
+    };
+
+    /**
+     * Listen keydown events while this component is focused overring the default native input behavior.
+     * Only digits and backspace are allowed.
```
[the remark this answers is not in the document]
Comment (thr0wn) [reply src/components/InputNumber.js:58]: Hi @obiyankenobi, this component tries to mimic an atm input, so the cursor isn't considered. You are right, only the end of the text input changes here. An alternative way to not ignore the cursor position is to do something like below, but I would not recommend making a hybrid of atm and 'cursor aware' behavior.
1*jm0Qb7D7xYJFbJeoO1W33A.

File: src/components/InputNumber.js [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+      const value = rawValue.replace(/[^\d]/g, "").padStart(precision + 1, "0");
+      const decimalPart = Intl.NumberFormat(locale).format(
+        value.substr(0, value.length - precision)
+      );
+      const fractionalPart = value.substr(value.length - precision);
+      return `${decimalPart}${separator}${fractionalPart}`;
+    };
+
+    /**
+     * Listen keydown events while this component is focused overring the default native input behavior.
```
[the remark this answers is not in the document]
Comment (thr0wn) [reply src/components/InputNumber.js:57]: Thanks! 😅

**Comment by thr0wn** [comment]
The new input component works like an atm:


It reacts to paste events too:

File: src/components/InputNumber.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (thr0wn) [reply src/components/InputNumber.js:59]: Delete is now accepted, thanks!

[the remark this answers is not in the document]
Comment (obiyankenobi) [reply src/components/InputNumber.js:59]: I actually don't mind not accepting delete, it's just a matter of making sure the user can't change the cursor. What did you think was better during your tests?

[the remark this answers is not in the document]
Comment (obiyankenobi) [reply src/components/InputNumber.js:59]: Ok, so I understand now that the delete/backspace will always remove the last number. For me it's ok.

**Review by obiyankenobi — APPROVED** [review]

**Comment by thr0wn** [comment]
New improvements:

<table>
  <thead>
    <tr>
      <th>Caret at the end</td>
      <th>Delete all (Ctrl+A or Command + A) and Delete
</td>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>

      </td>
      <td>
        
      </td>
    </tr>
  </tbody>
</table>

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**